[EDU-1654] - Add generate Ably token authentication example (#2322)
ably/docs: examples/auth-generate-token/page.md

> # Generate an Ably token to authenticate clients

Reviewer: Should this be "request" rather than "generate" (and generate is for JWT?)

Also "Ably Token" needs to be capitalised throughout (sorry)

> Use JWT tokens to securely authenticate users within your application.

Reviewer: "JWT tokens" or "JWTs"?

> Use an Ably token to securely authenticate users within your application.

Reviewer: ```suggestion
Use an Ably token to securely authenticate your users.
```

> JWT tokens enable you to grant users access to Ably's real-time messaging services. They are most commonly used to issue time-limited credentials that securely authorize users to connect to Ably channels.

Reviewer: ```suggestion
JWT tokens enable you to grant users access to the Ably platform. They are used to issue time-limited credentials that securely authorize users, and manage which resources each user has access to.
```

> JWT tokens are an essential component of applications that require secure and scalable real-time communication. Before any application can use any of the Ably products whether they be Pub/Sub, Spaces, Chat, or LiveSync, they'll need to first authenticate.

Reviewer: ```suggestion
The first step for an application to communicate with Ably requires them authenticating.
```

I think we should replace the deleted info here with explaining about authUrl and token renewal. WDYT?

> Authentication is implemented using [Ably core Pub/Sub](https://ably.com/docs/auth). The Pub/Sub SDK contains a set of purpose-built APIs that abstract away the complexities involved in architecting websocket features. It is built on top of Ably's core platform, and so it provides the same performance guarantees and scaling potential.

Reviewer: I think we need to reword this to state that authentication is handled by Pub/Sub and how that bleeds into the other products.

> Authentication is implemented using [Ably core Pub/Sub](https://ably.com/docs/auth). The Pub/Sub SDK provides authentication capabilities that are shared across all Ably products. The SDK includes purpose-built APIs that streamline pub/sub features while delivering Ably's performance guarantees and scaling capabilities.

Reviewer: ```suggestion
Authentication is implemented using [Ably core Pub/Sub](https://ably.com/docs/auth). The Pub/Sub SDK provides the authentication mechanism that is utilized by all Ably products.
```

Then finish it off with whatever line we agree on regarding the Pub/Sub SDK (`The Pub/Sub SDK provides a set of flexible APIs capable of building any realtime application. It is powered by Ably's reliable and scalable platform.`) was my suggestion in the other PR.

## Resources

// React brief (Only visible if viewing the React example)

> This example uses a frontend and a backend client. The frontend is a React application and the backend is an Express Node.js application. Both of these applications use the [Ably JS SDK](https://github.com/ably/ably-js).

Reviewer: ```suggestion
This example uses a frontend and a backend client. The frontend is a React application and the backend is an Express Node.js application. Both of these applications use Ably Pub/Sub.
```

The server application calls [`auth.requestToken();`](https://ably.com/docs/auth/token#ably-token) to request a token from Ably in order to pass it to the client to be used as authentication. The client calls the server's publicly exposed endpoint `/auth-url` to request this token.

Find out more about [authentication](https://ably.com/docs/auth).

// End React brief

// Javascript brief (Only visible if viewing the Javascript example)

> This example uses a frontend and a backend client. The frontend is a Typescript application and the backend is an Express Node.js application. Both of these applications use the [Ably JS SDK](https://github.com/ably/ably-js).

Reviewer: ```suggestion
This example uses a frontend and a backend client. The frontend is a Typescript application and the backend is an Express Node.js application. Both of these applications use Ably Pub/Sub.
```

The server application calls [`auth.requestToken();`](https://ably.com/docs/auth/token#ably-token) to request a token from Ably in order to pass it to the client to be used as authentication. The client calls the server's publicly exposed endpoint `/auth-url` to request this token.

Find out more about [authentication](https://ably.com/docs/auth).

// End Javascript brief

> ## View on Github

Reviewer: You need to add an additional step to these instructions to cover adding your API key to the server.

Author: I'm sure I added a comment here but I can't find it... Steps 5-10 are these steps.
